app/models/event.rb

```diff
@@ -3,6 +3,24 @@ class Event < ApplicationRecord
   validates_presence_of :description
   validates_presence_of :eventType
   has_many :event_statuses
-  has_many :users, :through => :event_statuses
+  has_many :users, :through => :event_statuses, validate: false
+  validate :check_users
+  validate :check_preferences
   enum eventType: %i[event info message]
+
+  private
+  def check_users
+    if self.users.blank?
+      errors.add(:users, I18n.t('global.error_users'))
+    end
+  end
+
+  def check_preferences
+    if !self.use_email? && !self.use_call? && !self.use_text?  && !self.use_app?
```

Comment:
Is `self._` referring to a user object here, or the event object? It's in the event model. I know those methods belong to the user now, so I am a bit confused as to how this works